pkg/observer/controller.go

```diff
@@ -309,6 +370,15 @@ func (c *controller) Run() {
 	klog.V(3).Infof("waiting for %v to allow changes to settle", c.WaitInterval)
 	select {
 	case <-time.After(c.WaitInterval):
+		timeout := 0
+		retries := 20
+		for !c.safeToStartCycle() {
+			time.Sleep(10 * time.Second)
+			timeout++
+			if timeout >= retries {
+				klog.Fatalln("Not safe to start CNR. Retries exceeded", retries)
```

Comment:
Thoughts about logging Error and `return`ing thus skipping the create this loop vs `Fatal`? 

I tend to reserve Fatal for unrecoverable circumstances where continuing doesn't make sense, or could cause damage.